lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for Yoda conditions, i.e. comparison operations where
+      # readability is reduced because the operands are not ordered the same
+      # way as they would be ordered in spoken English.
+      #
+      # @example
+      #
+      #   # bad
+      #   99 == foo
+      #   "bar" == foo
+      #
+      # @example
+      #
+      #   # good
+      #   foo == 99
+      #   foo == "bar"
+      class YodaCondition < Cop
+        MSG = 'Reverse the order in conditional statement: `%s` == `%s`.'.freeze
```

Comment:
The comparison method can be something other than `==`. I think you could change this to `MSG = 'Reverse the order in conditional statement: `%s`.'.freeze`, and pass `node.source` into format.